### ISSUE TITLE
Remove `--headless` command line argument from Electron

### ISSUE THREAD

**Issue by ckerr** [electron/electron#27052]
From discussion with @jkleinsc (and indirectly @nornagon), it looks like if you want to run a headless Electron e.g. for CI testing, in Docker, etc. the approach is to use a virtual framebuffer e.g. xvfb and run it that way. That we allow the `--headless` argument seems to confuse users, as it implies that we support it.

xref https://github.com/electron/electron/issues/26974

**Comment by nornagon** [comment]
FWIW, I think it'd be cool if we _did_ support `--headless`, but it's likely a nontrivial amount of work. `xvfb` is a good workaround for now, but I'd be happy to review PRs if anyone wants to take a stab at "properly" supporting headless mode.

**Comment by trevyn** [comment]
@nornagon I agree, I have a desire for proper headless mode as well -- my app provides a web server that a user might want to run in headless mode on a remote machine.

Was this PR along the right lines? https://github.com/electron/electron/pull/26278

**Comment by colin969** [comment]
Seconding removing this if it isn't supported.

The amount of time I've wasted on Electron debugging over the years because of weird undocumented behaviour in command line switches, how certain things are positional etc, is criminal.
